app/Http/Middleware/BiblesAllApiKeyCompat.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace App\Http\Middleware;
+
+use Closure;
+
+/**
+   * Paginates the bibles.all endpoint depending on api key 
+   * (backward compat for bibleis and gideons)
+   *
+   * @param  \Illuminate\Http\Request  $request
+   * @param  \Closure  $next
+*/
+class BiblesAllApiKeyCompat
+{
+    public function handle($request, Closure $next)
+    {   
+        $compat_api_keys = config('auth.compat_users.api_keys');
```

Comment:
can you determine the route at this point? If yes, please check that the route is v4_bibles.all before applying this logic. I suspect there will be other such logic that we need to implement for bible.is and gideon until they update their code